extension/extension.cpp

```diff
@@ -319,6 +319,10 @@ DETOUR_DECL_MEMBER9(CBaseServer__ConnectClient, IClient*, netadr_t&, address, in
 	EBeginAuthSessionResult result = BeginAuthSession(pvTicket, cbTicket, g_lastClientSteamID);
 	if (result != k_EBeginAuthSessionResultOK)
 	{
+		// Explicitly end the auth session before rejecting to prevent memory leak
+		EndAuthSession(g_lastClientSteamID);

```

Comment:
when we call rejectconnection it already ends auth session here:
- https://github.com/srcdslab/sm-ext-connect/blob/16066e788d1ec49383030f15a3a4c937cf3e677c/extension/extension.cpp#L353